lib/measured/unit.rb

```diff
@@ -56,8 +56,8 @@ def inverse_conversion_amount
 
   private
 
-  def case_insensitive(comparison)
-    [comparison].flatten.map(&:downcase)
+  def downcased_names
+    @downcased_names ||= @names.map(&:downcase)
```

Comment:
I think if you make this a set rather than an array, it'll be much faster... I benchmarked set vs array's `include?`:

```bash
  4.370000   0.010000   4.380000 (  4.388877)  # Array.include? with 1000 elements
  0.010000   0.000000   0.010000 (  0.012523)  # Set.include? with 1000 elements
```

My test code was this:


```ruby
require 'set'
require 'benchmark'


random_array = 1000.times.map{(0...8).map { (65 + rand(26)).chr }.join}
sample = random_array.last
random_set = Set.new random_array

puts Benchmark.measure {
  a = 0
  while a < 100000
    raise 'bad' unless random_array.include?(sample)
    a += 1
  end
}

puts Benchmark.measure {
  a = 0
  while a < 100000
    raise 'bad' unless random_set.include?(sample)
    a += 1
  end
}
```